### PR TITLE
Remove function marked as deprecated until 3.0.0

### PR DESCRIPTION
**Pull request by cibernox**
IDK if this is here intentionally or it's just an oversight, but it should be safe to remove it now

### CODE REVIEW

**Comment by locks** [comment]
I think you still need to remove it from the `Ember` namespace building code: https://travis-ci.org/emberjs/ember.js/jobs/348773814#L1108

**Comment by rwjblue** [comment]
We need to confirm this isn’t needed for supporting the Ember-2-legacy addon

**Comment by cibernox** [comment]
@rwjblue I revisited this and realized there was much more to it. Not it's fully removed.

**Review by rwjblue — CHANGES_REQUESTED** [review]
Thanks for working on this!  You have essentially removed the `Ember.required()` concept here (which is great!).  Unfortunately, we have committed to maintaining support for this through Ember 3.4 when using the `ember-2-legacy` addon. 

tldr; This PR needs to be held from merging until master is representing 3.5.

**Comment by cibernox** [comment]
It seems too entangled to port it to `ember-2-legacy`, so let's wait.

**Comment by krisselden** [comment]
required though does nothing because everything does applyPartial, if the addon just made it not error but do nothing it would likely not affect anything in the wild. This feature has not really worked since before 1.0, it only does something for apply() and everything else uses extend which uses applyPartial

**Comment by cibernox** [comment]
@krisselden you're saying thats if I add a dummy Ember.required in the addon we don't have to wait to remove it?

**Comment by cibernox** [comment]
@rwjblue @krisselden Since this feature wasn't really working this should not have any impact in the wild other than the lost of "Ember.required", so I added a "placebo polyfill" that returns an empty function: https://github.com/emberjs/ember-2-legacy/pull/66

**Comment by cibernox** [comment]
This feature has been "polyfilled" (kind-of) in the ember-2-legacy-addon, so this is ready for another review.

**Comment by cibernox** [comment]
@rwjblue I this on time for the new release?

File: packages/ember/tests/reexports_test.js [hunk 1]
```diff
@@ -256,6 +255,34 @@ let allExports = [
   ['ContainerDebugAdapter', 'ember-extension-support'],
 ];
 
-if (ENV._ENABLE_PROPERTY_REQUIRED_SUPPORT) {
-  allExports.push(['required', 'ember-metal']);
+allExports.forEach(reexport => {
```
Comment (rwjblue) [inline packages/ember/tests/reexports_test.js]: I think you have a rebase issue here, if you notice this loop is now included above on line 10.

Comment (cibernox) [reply packages/ember/tests/reexports_test.js]: Fixed

**Review by rwjblue — APPROVED** [review]
